File: SpotHunt/Factor.cs
public sealed class Factor {
	private readonly string name;
	private readonly double rating;
	private readonly string mathSymbol;
	
	private Factor(string name, double rating, string mathSymbol) {
		this.name = name;
		this.rating = rating;
		this.mathSymbol = mathSymbol;
	}

	public static Factor TPD = new Factor("Total Player Distance", 0.5, ">");
	public static Factor ST = new Factor("Surround Threat", 1.0, "<");
	public static Factor SD = new Factor("Spot Distance", 0.5, "<");
	public static Factor FDC = new Factor ("Fastest Danger Cost", 1.0, "<");
	public static Factor HD = new Factor("Highest Danger", 1.0, "<");

	public double getRating() {
		return rating;
	}
	
	public string getName() {
		return name;
	}
	
	public string getSymbol() {
		return mathSymbol;
	}
}

File: spothunt/Factor.java
package spothunt;

public enum Factor {
	TPD("Total Player Distance", 0.5, ">"),
	ST("Surround Threat", 1, "<"),
	SD("Spot Distance", 0.5, "<"),
	FDC("Fastest Danger Cost", 1, "<"),
	HD("Highest Danger", 1, "<");
	
	private String name;
	private double rating;
	private String mathSymbol;
	
	private Factor(String name, double rating, String mathSymbol) {
		this.name = name;
		this.rating = rating;
		this.mathSymbol = mathSymbol;
	}
	
	public double getRating() {
		return rating;
	}
	
	public String getName() {
		return name;
	}
	
	public String getSymbol() {
		return mathSymbol;
	}
}
